lib/srv/desktop/rdp/rdpclient/src/client.rs

```diff
@@ -303,221 +308,187 @@ impl Client {
             .take()
             .ok_or_else(|| ClientError::InternalError("function_receiver failed".to_string()))?;
 
-        let mut read_loop_handle = Client::run_read_loop(
+        let read_loop = Client::read_loop(
             self.cgo_handle,
             read_stream,
             self.x224_processor.clone(),
             self.client_handle.clone(),
         );
 
-        let mut write_loop_handle = Client::run_write_loop(
+        let write_loop = Client::write_loop(
             self.cgo_handle,
             write_stream,
             function_receiver,
             self.x224_processor.clone(),
             self.pending_resize.clone(),
         );
 
-        // Wait for either loop to finish. When one does, abort the other and return the result.
+        // Wait for either loop to finish. When one does, cancel the other and return the result.
         tokio::select! {
-            res = &mut read_loop_handle => {
-                // Read loop finished, abort the other tasks and return the result.
-                write_loop_handle.abort();
-                res?
-            },
-            res = &mut write_loop_handle => {
-                // Write loop finished, abort the other tasks and return the result.
-                read_loop_handle.abort();
-                match res {
-                    Ok(Ok(())) => Ok(None),
-                    Ok(Err(client_error)) => Err(client_error),
-                    Err(join_error) => Err(join_error.into()),
-                }
-            }
+            res = read_loop => res,
```

Comment:
This changes semantics, previously both loops run in parallel, now they are run concurrently on single thread. Are we completely sure that it's safe in all cases (i.e. there are no blocking calls anymore anywhere etc)?

Reply:
Blocking calls are still a problem even in the current implementation, just one that's masked if there's not too many of them at once - the spawned tokio tasks don't get a dedicated thread each, they still run as part of the runtime and only a handful of threads (one per core, by default) are available to run them.

Shouldn't we try to expose problems so we can fix them rather than attempt to maybe work around them in some not-well-determined cases? This PR adds a `spawn_blocking` call to `cgo_handle_rdp_connection_activated`, for example.

Reply:
Yeah, you might be right that it's better to expose the problem, I'd just go and check that all `cgo_` calls are wrapped